### PR TITLE
MGDAPI-6015 change 3scale to add topologySpreadConstrants via apimanager CR

### PR DESCRIPTION
**Pull request by austincunningham**
# Issue link

https://issues.redhat.com/browse/MGDAPI-6015

# What

when the mas-0.11.8 build comes out this needs to be merged

# Verification steps

- [ ] cherry-pick this commit to the mas-0.11.8 branch 
- [ ] Install rhoam locally
```bash
INSTALLATION_TYPE=managed-api LOCAL=false make cluster/prepare/local
INSTALLATION_TYPE=managed-api LOCAL=false USE_CLUSTER_STORAGE=true make deploy/integreatly-rhmi-cr.yml
INSTALLATION_TYPE=managed-api LOCAL=false make code/run
```
- [ ] Wait for the install to finish
- [ ] Check that the topologySpreadConstraint has been added to all the DC
```bash
for dc in $(oc get deploymentconfigs --namespace=redhat-rhoam-3scale | grep config | awk '{print $1}'); do echo $dc "===================================>"; oc get dc $dc -oyaml | yq '.spec.template.spec.topologySpreadConstraints' ; done
```
expected output
```bash
apicast-production ===================================>
- labelSelector:
    matchLabels:
      app: 3scale-api-management
  maxSkew: 1
  topologyKey: topology.kubernetes.io/zone
  whenUnsatisfiable: ScheduleAnyway
apicast-staging ===================================>
- labelSelector:
    matchLabels:
      app: 3scale-api-management
  maxSkew: 1
  topologyKey: topology.kubernetes.io/zone
  whenUnsatisfiable: ScheduleAnyway
backend-cron ===================================>
- labelSelector:
    matchLabels:
      app: 3scale-api-management
  maxSkew: 1
  topologyKey: topology.kubernetes.io/zone
  whenUnsatisfiable: ScheduleAnyway
backend-listener ===================================>
- labelSelector:
    matchLabels:
      app: 3scale-api-management
  maxSkew: 1
  topologyKey: topology.kubernetes.io/zone
  whenUnsatisfiable: ScheduleAnyway
backend-worker ===================================>
- labelSelector:
    matchLabels:
      app: 3scale-api-management
  maxSkew: 1
  topologyKey: topology.kubernetes.io/zone
  whenUnsatisfiable: ScheduleAnyway
system-app ===================================>
- labelSelector:
    matchLabels:
      app: 3scale-api-management
  maxSkew: 1
  topologyKey: topology.kubernetes.io/zone
  whenUnsatisfiable: ScheduleAnyway
system-memcache ===================================>
- labelSelector:
    matchLabels:
      app: 3scale-api-management
  maxSkew: 1
  topologyKey: topology.kubernetes.io/zone
  whenUnsatisfiable: ScheduleAnyway
system-searchd ===================================>
- labelSelector:
    matchLabels:
      app: 3scale-api-management
  maxSkew: 1
  topologyKey: topology.kubernetes.io/zone
  whenUnsatisfiable: ScheduleAnyway
system-sidekiq ===================================>
- labelSelector:
    matchLabels:
      app: 3scale-api-management
  maxSkew: 1
  topologyKey: topology.kubernetes.io/zone
  whenUnsatisfiable: ScheduleAnyway
zync ===================================>
- labelSelector:
    matchLabels:
      app: 3scale-api-management
  maxSkew: 1
  topologyKey: topology.kubernetes.io/zone
  whenUnsatisfiable: ScheduleAnyway
zync-database ===================================>
- labelSelector:
    matchLabels:
      app: 3scale-api-management
  maxSkew: 1
  topologyKey: topology.kubernetes.io/zone
  whenUnsatisfiable: ScheduleAnyway
zync-que ===================================>
- labelSelector:
    matchLabels:
      app: 3scale-api-management
  maxSkew: 1
  topologyKey: topology.kubernetes.io/zone
  whenUnsatisfiable: ScheduleAnyway
```

### CODE REVIEW

File: pkg/products/threescale/reconciler.go [hunk 1]
```diff
@@ -2896,9 +2921,7 @@ func (r *Reconciler) reconcileDeploymentConfigs(ctx context.Context, serverClien
 			ctx,
 			serverClient,
 			resources.SelectFromDeploymentConfig,
-			resources.AllMutationsOf(
-				resources.MutateZoneTopologySpreadConstraints("app"),
```
Comment (MStokluska) [inline pkg/products/threescale/reconciler.go:2900]: is the entire `reconcileDeploymentConfigs` even required anymore? I'm not sure I'm missing something but it doesn't seem to be doing anything atm?

Comment (austincunningham) [reply pkg/products/threescale/reconciler.go:2900]: Agree don't think its doing anything without the mutate , but like you I'm not sure I am missing something.

Comment (austincunningham) [reply pkg/products/threescale/reconciler.go:2900]: removed `reconcileDeploymentConfigs`

**Comment by trepel** [comment]
I validated this by following the verification steps on both fresh install and upgrade. The topologySpreadConstraint were present as expected. Feel free to lgtm and approve this once this is unhold

**Pull request closed without merging**